utils/regression-analysis/src/main.rs

```diff
@@ -200,23 +214,223 @@ fn compare<P: AsRef<Path>>(data_path: P, current_junit_path: P, disable_filter:
+            }
+        ) => {{
+            // check field is exist
+            let $name::<gear_runtime::Runtime> {
+                $( $field: _, )+
+            } = Default::default();
+
+            $(
+                let field = add_weights!(@field $weights $field $( : $underscore )?);
+                $benches.extend(field);
+            )+
+        }};
+        (@field $weights:ident $field:ident: _) => { None };
+        (@field $weights:ident _phantom) => { None };
+        (@field $weights:ident $field:ident) => {
+            $weights.get(stringify!($field)).map(|weight| {
+                GithubActionBenchmark {
+                    name: stringify!($field).to_string(),
+                    unit: "ns".to_string(),
+                    value: weight.calc_weight() / 1000,
+                    range: None,
+                    extra: None,
+                }
+            })
+        };
+    }
+
+    let file = fs::File::open(input_file).unwrap();
+    let map: HashMap<String, WeightBenchmark> = serde_json::from_reader(file).unwrap();
+    let map: HashMap<String, WeightBenchmark> = map
+        .into_iter()
+        .map(|(name, bench)| {
+            (
+                // we strip prefix because WASM instruction benchmarks have it
+                name.strip_prefix("instr_")
+                    .map(|x| x.to_string())
+                    .unwrap_or(name),
+                bench,
+            )
+        })
+        .collect();
+
+    let mut benches = vec![];
+
+    match kind {
+        WeightsKind::HostFn => {
+            add_weights! {
+                weights = map;
+                benches = benches;
+                HostFnWeights {
+                    _phantom,
+                    alloc,
+                    gr_gas_available,
+                    gr_msg_id,
+                    gr_origin,
+                    gr_program_id,
+                    gr_source,
+                    gr_value,
+                    gr_value_available,
+                    gr_size,
+                    gr_read,
+                    gr_read_per_byte,
+                    gr_block_height,
+                    gr_block_timestamp,
+                    gr_send_init,
+                    gr_send_push,
+                    gr_send_push_per_byte,
+                    gr_send_commit,
+                    gr_send_commit_per_byte,
+                    gr_reply_commit,
+                    gr_reply_commit_per_byte,
+                    gr_reply_push,
+                    gr_reply_push_per_byte,
+                    gr_reply_to,
+                    gr_debug,
+                    gr_exit_code,
+                    gr_exit,
+                    gr_leave,
+                    gr_wait,
+                    gr_wake,
+                    gr_create_program_wgas,
+                    gr_create_program_wgas_per_byte,
+                    gas,
+                }
+            }
+        }
+        WeightsKind::Instruction => {
```

Comment:
We also have `Extrinsics` that should be one more variant of `WeightsKind`

Reply:
What `Extrinsics`?

Reply:
`pallet-gear`'s calls: `send_message`, `upload_program` etc. They have also benchmarked weights